packages/flutter/lib/src/rendering/editable.dart

```diff
@@ -742,178 +738,804 @@ class RenderEditable extends RenderBox with RelayoutWhenSystemFontsChangeMixin {
     return 0;
   }
 
-  void _handleMovement(
```

Comment:
To explain this diff: The `_handleMovement` method has been removed and is now split into all of the separate methods below.